src/Tasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -267,6 +269,112 @@ internal static FrameworkName GetTargetFrameworkAttribute(string path)
             return false;
         }
 
+#if !FEATURE_ASSEMBLYLOADCONTEXT
+        /// <summary>
+        /// Collects the metadata and attributes for specified assembly.
+        /// The requested properties are used by legacy project system.
+        /// </summary>
+        internal AssemblyAttributes GetAssemblyMetadata()
+        {
+            IntPtr asmMetaPtr = IntPtr.Zero;
+            ASSEMBLYMETADATA asmMeta = new();
+            try
+            {
+                IMetaDataImport2 import2 = (IMetaDataImport2)_assemblyImport;
+                _assemblyImport.GetAssemblyFromScope(out uint assemblyScope);
```

Comment:
Does this require opening and rescanning every assembly again, after RAR already did so? Is that avoidable?

Reply:
It was a conscious decision to do this in a separate task to avoid bloating RAR and also to mitigate risk by keeping the old and new logic as much 1:1 as possible. It is certainly avoidable and we could explore what it would look like if we moved this work to RAR and eliminated the new task.

@YuliiaKovalova can you please check if there is overlap between the data calculated here and what's already returned by RAR?

Reply:
For example, now RAR returns this data for mscorlib
![image](https://github.com/dotnet/msbuild/assets/95473390/da50db11-d64c-46e6-a222-724467781de5)
But we need these attributes too
![image](https://github.com/dotnet/msbuild/assets/95473390/78c2319f-720e-4e2c-a302-85eed837dfab)